blockmanager.go

```diff
@@ -1334,11 +1334,37 @@ func (b *blockManager) detectBadPeers(headers map[string]*wire.MsgCFHeaders,
 
 	log.Warnf("Detected cfheader mismatch at height=%v!!!", targetHeight)
 
-	// Get the block header for this height, along with the block as well.
+	// Get the block header for this height.
 	header, err := b.server.BlockHeaders.FetchHeaderByHeight(targetHeight)
 	if err != nil {
 		return nil, err
 	}
+
+	// Fetch filters from the peers in question.
+	// TODO(halseth): query only peers from headers map.
+	filtersFromPeers := b.fetchFilterFromAllPeers(
+		targetHeight, header.BlockHash(), fType,
+	)
+
+	var badPeers []string
+	for peer, _ := range headers {
+		_, ok := filtersFromPeers[peer]
+
+		// If a peer did not respond, ban it immediately.
+		if !ok {
+			log.Warnf("Peer %v did not respond to filter "+
+				"request, considering bad.", peer)
+			badPeers = append(badPeers, peer)
+			continue
+		}
+	}
+
+	if len(badPeers) != 0 {
+		return badPeers, nil
```

Comment:
Any reason we want to return here and not continue to determine if we have any other bad peers?

Reply:
Idea is to return as early as possible, before starting expensive operations such as getting blocks.